src/main/java/in/venkat/service/ShowService.java

```diff
@@ -0,0 +1,204 @@
+	 * 
+	 * @param filmGenre
+	 * @param filmLanguage
+	 * @return
+	 * @throws SQLException
+	 * @throws InvalidDetailsException
+	 * @throws InvalidNameException
+	 * @throws EmptyFieldException
+	 * @throws DbException
+	 * @throws Exception
+	 */
+	public static List<Show> searchContents(String filmGenre, String filmLanguage) {
+		boolean isValid;
+		List<Show> search = null;
+
+		try {
+			isValid = ValidateSearchDetails.validateDetails(filmGenre, filmLanguage);
+			if (isValid) {
+				search = searchMoviesByLanguageAndGenre(filmGenre, filmLanguage);
+			}
+		} catch (EmptyFieldException | InvalidNameException | InvalidDetailsException e1) {
+			Logger.log(e1.getMessage());
+		}
+
+		return search;
+	}
+
+	/**
+	 * 
+	 * @param filmGenre
+	 * @param filmLanguage
+	 * @return
+	 * @throws SQLException
+	 * @throws DbException
+	 * @throws Exception
+	 */
+	public static List<Show> searchMoviesByLanguageAndGenre(String filmGenre, String filmLanguage) {
+		List<Show> primeMovieList;
+		List<Show> filteredMovieList = new ArrayList<>();
+
+		try {
+			primeMovieList = ShowListDao.getShowDetails();
+			filteredMovieList.removeAll(filteredMovieList);
+			for (Show show : primeMovieList) {
+				if (show.getMovieGenre().equalsIgnoreCase(filmGenre.trim())
+						&& show.getMovieLanguage().equalsIgnoreCase(filmLanguage.trim()))
+					filteredMovieList.add(show);
+			}
+		}
+
+		catch (DbException e) {
+			Logger.log(e.getMessage());
+		}
+
+		return filteredMovieList;
+
+	}
+
+	/**
+	 * This method is used to search movies by membership
+	 * 
+	 * @param membership
+	 * @return
+	 * @throws ClassNotFoundException
+	 * @throws SQLException
+	 * @throws DbException
+	 * @throws Exception
+	 */
+	public static List<Show> searchMoviesByMembership(String membership) {
+		List<Show> primeMovieList;
+		List<Show> filteredMovieList = new ArrayList<>();
+		try {
+			primeMovieList = ShowListDao.getShowDetails();
+			filteredMovieList.removeAll(filteredMovieList);
+			for (Show show : primeMovieList) {
+				if (show.getMembership().equalsIgnoreCase(membership.trim())) {
+					filteredMovieList.add(show);
+					Logger.log(show);
+				}
+			}
+		} catch (DbException e) {
+			Logger.log(e.getMessage());
+		}
+
+		return filteredMovieList;
+
+	}
+
+	/**
+	 * This method is used to search movies by the released year
+	 * 
+	 * @param year
+	 * @return
+	 * @throws ClassNotFoundException
+	 * @throws SQLException
+	 * @throws DbException
+	 * @throws Exception
+	 */
+	public static List<Show> searchMoviesByYear(int year) {
+		List<Show> primeMovieList;
+		List<Show> filteredMovieList = new ArrayList<>();
+
+		try {
+			primeMovieList = ShowListDao.getShowDetails();
+			filteredMovieList.removeAll(filteredMovieList);
+			for (Show show : primeMovieList) {
+				if (show.getMovieYear() == year) {
+					filteredMovieList.add(show);
+					Logger.log(show);
+
+				}
+
+			}
+		} catch (DbException e) {
+			Logger.log(e.getMessage());

```

Comment:
Exception should not be suppressed.

Reply:
resolved this by logger.exception method

Reply:
not  suppressed.